src/alchemlyb/tests/parsing/test_amber.py

```diff
@@ -25,13 +25,16 @@ def invalid_file(request):
                           for filename in leg])
 def test_dHdl(filename,
               names=('time', 'lambdas'),
-              shape=(500, 1)):
+              shape=(500, 1),
+              first_time=22.0,
+              last_time=1020.0):
     """Test that dHdl has the correct form when extracted from files."""
     dHdl = extract_dHdl(filename, T=300)
 
     assert dHdl.index.names == names
     assert dHdl.shape == shape
-
+    assert dHdl.index.values[0][0] == first_time
+    assert dHdl.index.values[-1][0] == last_time
```

Comment:
You could just add a new test that checks for dHdl.index.values[0][0] and dHdl.index.values[-1][0]

Make sure to use `np.isclose` to check float point numbers.

Reply:
I've done so now, using ```isclose``` and also testing for the correct reading of u_nk values, as I had forgotten that previously!